ansible/roles/airflow/tasks/main.yml

```diff
@@ -0,0 +1,155 @@
+    group: "{{ item.login }}"
+    uid: "{{ item.id }}"
+    comment: "{{ item.comment }}"
+    shell: /usr/sbin/nologin
+    createhome: no
+    home: /nonexistent
+    state: present
+  with_items: "{{ svc_passwd }}"
+
+- name: shared tmp directory to workaround lack of tmpfs support in docker_container
+  file: dest=/srv/tmp state=directory owner=root group=root mode=0750
+
+- name: config file directory
+  file: dest=/srv/etc state=directory owner=root group=root mode=0750
+
+- name: docker network for airflow
+  docker_network:
+    name: af
+    driver_options:
+      com.docker.network.bridge.name: braf
+    ipam_options:
+      subnet: 172.26.43.0/24
+      gateway: 172.26.43.1
+
+- name: redis .../etc directory
+  file: dest=/srv/etc/af-redis state=directory owner=root group=root mode=0755
+- name: redis .../etc/redis.conf
+  template: dest=/srv/etc/af-redis/redis.conf src=af-redis.conf mode=0640 owner=root group=afredis
+- name: redis /data directory
+  file: dest=/srv/af-redis state=directory owner=afredis group=afredis mode=0750
+- name: docker run redis
+  docker_container:
+    image: redis:3.2
+    name: af-redis
+    hostname: af-redis
+    networks: [{name: af}]
+    purge_networks: true # drop `bridge` network
+    volumes:
+      - /srv/etc/af-redis:/usr/local/etc:ro
+      - /srv/af-redis:/data:rw
+    command: /usr/local/etc/redis.conf
+    user: "{{ passwd.afredis.id}}:{{ passwd.afredis.id}}"
+    restart_policy: on-failure
+    restart_retries: 2
+
+# 1. if `initdb` fails read `Arbitrary --user Notes` at https://hub.docker.com/_/postgres/
+# 2. /run has to be writable by postgres and `docker_container` does not support tmpfs
+#    https://github.com/ansible/ansible/pull/22646
+- name: airflow postgres tmp directory
+  file: dest=/srv/tmp/af-psql state=directory owner=afpsql group=afpsql mode=0750
+- name: airflow postgres /data directory
+  file: dest=/srv/af-psql state=directory owner=afpsql group=afpsql mode=0750
+- name: docker run airflow postgres
+  docker_container:
+    image: postgres:9.6
+    name: af-psql
+    hostname: af-psql
+    networks: [{name: af}]
+    purge_networks: true # drop `bridge` network
+    volumes:
+      - /srv/tmp/af-psql:/run/postgresql:rw
+      - /srv/af-psql:/srv/af-psql:rw
+    env:
+      PGDATA: /srv/af-psql
+      POSTGRES_USER: airflow
+      POSTGRES_PASSWORD: "{{ airflow_postgres_password }}"
+    user: "{{ passwd.afpsql.id }}:{{ passwd.afpsql.id }}"
+    restart_policy: on-failure
+    restart_retries: 2
+
+- name: shared airflow config templates
+  file: dest=/srv/etc/af-share-{{ airflow_tag }} state=directory owner=root group=root mode=0750
+- stat: path=/srv/etc/af-share-{{ airflow_tag }}/passwd
+  register: afpasswd
+- stat: path=/srv/etc/af-share-{{ airflow_tag }}/group
+  register: afgroup
+- stat: path=/srv/etc/af-share-{{ airflow_tag }}/airflow.cfg
+  register: afcfg
+
+- name: create temporary openobservatory/airflow container to extract file
```

Comment:
Is this really necessary or perhaps I don't understand why this should in this way.